openmc/mgxs/mdgxs.py

```diff
@@ -605,20 +655,45 @@ def print_xs(self, subdomains='all', nuclides='all', xs_type='macro'):
 
                     template = '{0: <12}Group {1} [{2: <10} - {3: <10}eV]:\t'
 
-                    # Loop over energy groups ranges
-                    for group in range(1, self.num_groups+1):
-                        bounds = self.energy_groups.get_group_bounds(group)
-                        string += template.format('', group, bounds[0], bounds[1])
-                        average = self.get_xs([group], [subdomain], [nuclide],
-                                              xs_type=xs_type, value='mean',
-                                              delayed_groups=[delayed_group])
-                        rel_err = self.get_xs([group], [subdomain], [nuclide],
-                                              xs_type=xs_type, value='rel_err',
-                                              delayed_groups=[delayed_group])
-                        average = average.flatten()[0]
-                        rel_err = rel_err.flatten()[0] * 100.
-                        string += '{:.2e} +/- {:1.2e}%'.format(average, rel_err)
-                        string += '\n'
+                    average_xs = self.get_xs(nuclides=[nuclide],
+                                             subdomains=[subdomain],
+                                             xs_type=xs_type, value='mean',
+                                             delayed_groups=[delayed_group])
+                    rel_err_xs = self.get_xs(nuclides=[nuclide],
+                                             subdomains=[subdomain],
+                                             xs_type=xs_type, value='rel_err',
+                                             delayed_groups=[delayed_group])
+                    rel_err_xs = rel_err_xs * 100.
+
+                    if self.num_polar > 1 or self.num_azimuthal > 1:
+                        # Loop over polar, azimuthal, and energy group ranges
+                        for pol in range(len(polar_bins) - 1):
+                            pol_low, pol_high = polar_bins[pol: pol + 2]
+                            for azi in range(len(azimuthal_bins) - 1):
+                                azi_low, azi_high = azimuthal_bins[azi: azi + 2]
+                                string += '\t\tPolar Angle: [{0:5f} - {1:5f}]'.format(
+                                    pol_low, pol_high) + \
+                                    '\tAzimuthal Angle: [{0:5f} - {1:5f}]'.format(
+                                    azi_low, azi_high) + '\n'
+                                for group in range(1, self.num_groups + 1):
+                                    bounds = \
+                                        self.energy_groups.get_group_bounds(group)
+                                    string += '\t' + template.format('', group,
```

Comment:
- [x]

Plus ive tested in some local files, though less so for MDGXS items.  There is nothing in the test suite to test these just yet. Last time through I said I was separately working on some stuff to better generate mgxs data specifically for testing purposes, so I figured I'd formally include this in the test suite then.

No example yet, as this is pretty niche.